Level 2

AnKiir opened the pull request:
Main page. Sign-up, sign-in, profile with exit. List of movies and option to view favored movies separately. Small animation on main page just to make it slightly less static.

Выбран макет Light-2.
Пожалуйста, дайте отдельно знать если где-то потеряны фоны (я физически не различаю белый и светлые оттенки серого).
В Фигме на странице авторизации/регистрации/профиля/404 нет футера. Добавлен на страницы для целостности.

AnKiir committed:
General functionality

Main page. Sign-up, sign-in, profile with exit. List of movies and option to view favored movies separately. Small animation on main page just to make it slightly less static.

AnKiir committed:
Update Profile.js

Small error fix.

AliaksandrHrychonak commented:
# Резюме по работе:

## Верстка и JSX

### Критерии, влияющие на работоспособность

- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
    - В роутах `/signin, /signup, /404, /profile` нет `<footer>` по макету
    - **На 1280px не по макету:**
    - https://disk.yandex.com/d/kOdw8SrNiFoYdw
    - https://disk.yandex.com/i/NgFZTbT64nXyrQ
    - https://disk.yandex.com/i/ynVgpX4JDtJswQ
    - https://disk.yandex.com/d/0RXaI0cPKCzQDQ
    - https://disk.yandex.com/d/8P4n5KNra6lK5g
    - https://disk.yandex.com/d/hpj8vEkhbhRKXw
    - https://disk.yandex.com/d/nOFropU5vjVY2Q
    - **На 768px не по макету:**
    - https://disk.yandex.com/d/p8t42N18zcXHXw
    - https://disk.yandex.com/d/fOSf4J5GTOmWrg
    - https://disk.yandex.com/d/4ZmmP3NQx7uqOw
    - **На 320px не по макету:**
    - https://disk.yandex.com/d/ejZRnCzx5fHc3w
    - https://disk.yandex.com/d/TFZOg6Yju4Y6VQ
    - https://disk.yandex.com/i/I5qrKrdICGQS6Q
    - https://disk.yandex.com/d/nl5TA_V-1m2seQ
    
    >Для более точного контроля отображения вёрстки можно воспользоваться плагином `PerfectPixel` для `Chrome`, можно в фигме выделить весь >макет, экспортировать его картинкой и наложить на вёрстку в браузере. Плагин позволяет наложить изображение на вёрстку в браузере, >тем самым можно слегкостью подправить недочёты по отображению.

- [ ] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
    - Не следует скрывать содержимое страниц от пользователя, привело `overflow: hidden;` следует удалить => `#root`
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [ ] В проекте есть:
    - [ ] <!--3.38--> директория images/ с картинками,
        - Папку необходимо переименовать
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
    - Всем интерактивным элементам(ссылки, кнопки, чекбоксы, лого) нужен эффект при наведении и выставленный курсор, проконтролируйте данный критерий для всех интерактивных элементов в проекте
        - например не хватает для `лого search-form__label-tumbler` 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [ ] <!--2.14--> Иконки экспортированы в формате SVG.
    - `hero-background.png` необходимо добавить в векторном формате
    - `logo.png` используйте векторный формат лого в проекте
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
    - Все поля инпут проекте кроме полей профиля нужно сделать обязательными к заполнению
    - Всем инпутам проекта нужно добавить плейсхолдеры
    - Поисковой инпут нужно сделать обязательным к заполнению
    - Всем полям **Имя и Пароль** нужна валидация минимальной/максимальной длинны значения
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [ ] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
    - `about-me__photo` альт необходимо заполнить подробнее
    - Зачем писать alt? https://habr.com/ru/sandbox/144678/
- [ ] <!--1.67--> Растровые и векторные изображения оптимизированы;
    - `/images` имеет вес `1.7мб`, изображения следует оптимизировать
    
## Количество баллов: **64**

AnKiir committed:
Правки, часть 1

Убран футер по макету. Оптимизированы картинки. переименованы папки. Иконки в svg. Убран overflow. Добавлены условия и заглушки для полей формы регистрации/авторизации, поисковой строки.

AnKiir committed:
Правки, часть 2

Отступы для разрешений 1280, 768 и 320 для главной страницы. Блок Студент имеет отличный от макета контент - равнение в выравнивании сделано по ссылке на Github в описании.

AnKiir committed:
Правки, часть 3

Отступы для внутренних страниц.

AliaksandrHrychonak commented:
# Резюме по работе:

## Верстка и JSX

### Критерии, влияющие на работоспособность

- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
   -  проект имеет ошибки валидации, удобно проверить валидатором https://validator.w3.org/#validate_by_input , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/TYwXvS7ppYMNZg
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
    - [x] <!--3.38--> директория images/ с картинками,
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
    - Всем интерактивным элементам(ссылки, кнопки, чекбоксы, лого) нужен эффект при наведении и выставленный курсор, проконтролируйте данный критерий для всех интерактивных элементов в проекте
        - например не хватает для `лого search-form__label-tumbler` 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [ ] <!--2.14--> Иконки экспортированы в формате SVG.
    - `hero-background.png` необходимо добавить в векторном формате
    - `logo.png` используйте векторный формат лого в проекте
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
    - Полям профиля нужно добавить плейсхолдеры
    - Полю **Имя** формы профиля нужна валидация минимальной/максимальной длинны значения
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [ ] <!--1.67--> Растровые и векторные изображения оптимизированы;
    - `about.png` имеет вес `1.6мб`, изображения следует оптимизировать
    
## Количество баллов: **89**
